SimG4CMS/Forward/interface/MtdSD.h

```diff
@@ -50,29 +47,23 @@ class MtdSD : public SensitiveTkDetector,
 
   void     Initialize(G4HCofThisEvent * HCE) override;
   void     EndOfEvent(G4HCofThisEvent * eventHC) override;
-  void     clear() override;
-  void     DrawAll() override;
   void     PrintAll() override;
 
-  double   getEnergyDeposit(const G4Step* step);
   void     fillHits(edm::PSimHitContainer&, const std::string&) override;
   void     clearHits() override;
   
-private:
-  void     update(const BeginOfJob *) override;
-  void     update(const BeginOfRun *) override;
+protected:
```

Comment:
why protected?
most probably all those classes can be declared ```final``` as well

Reply:
There many places like this one. I always try to keep "protected" if a method is "protected" in the base class. Concerning "final" - we do not use this normally, one can expect that "final" should be used in too many places instead of "override" but cannot be included automatically, so should done by hand. Making this in a small group of classes for me is not a good idea. Having "override" everywhere is , at least, consistent.

Reply:
I was suggesting ```final``` at class level

Reply:
@VinInn , this can be done but do we use "final" declaration for a class in other sub-packages of CMSSW?

Reply:
I do